petpal/comments/views.py

```diff
@@ -18,39 +18,51 @@
 SUCCESS:
 '''
 @api_view(['POST'])
+@permission_classes([IsAuthenticated])
 def comment_create_view(request):
-    if not request.user.is_authenticated:
-        return JsonResponse({'data': 'User not logged in'}, status=status.HTTP_401_UNAUTHORIZED)
-    is_author_seeker = request.data.get('is_author_seeker')
+    is_author_seeker = True if request.user.role == User.Role.SEEKER else False
     is_review = request.data.get('is_review')
     content = request.data.get('content')
-    seeker_email = request.data.get('seeker_email')
-    shelter_email = request.data.get('shelter_email')
+
+    # Check if email is seeker -> shelter, shelter -> seeker
+    if request.user.role == User.objects.get(email=request.data.get('recipient_email')).role:
```

Comment:
I have update the notion on this, remove seeker_email and shelter_email and change it to recipient_email